spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Input.scala

```diff
@@ -36,8 +36,16 @@ import scala.reflect.ClassTag
  * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
  */
 @SerialVersionUID(- 8525406230282608924L)
-class Input[T: ClassTag]()(implicit ev: TensorNumeric[T])
+class Input[T: ClassTag](val inputShape: Array[Int])(implicit ev: TensorNumeric[T])
```

Comment:
Don't use Array[Int], create a Shape class.

Reply:
What's the benefit for introducing another class? `case Shape(value: Array[Int])` ?

Reply:
For this specific purpose, use a Shape class will make the code more safe and easy to read.

Reply:
For nest shape, we can
1. not support it yet
2. use Shape(value: Array[ArrayIntOrShape])